src/EditorFeatures/CSharpTest/Completion/CompletionProviders/DeclarationNameCompletionProviderTests_NameDeclarationInfoTests.cs

```diff
@@ -0,0 +1,492 @@
+        public async void ModifierExclusion6()
+        {
+            var markup = @"
+class C
+{
+    override int $$
+}
+";
+            await VerifySymbolKinds(markup, SymbolKind.Method, SymbolKind.Property);
+        }
+
+        [Fact]
+        public async void ModifierExclusion7()
+        {
+            // Note that the async is not included in the incomplete member syntax
+            var markup = @"
+class C
+{
+    async int $$
+}
+";
+            await VerifySymbolKinds(markup, SymbolKind.Field, SymbolKind.Method, SymbolKind.Property);
+        }
+
+        [Fact]
+        public async void ModifierExclusion8()
+        {
+            // Note that the async is not included in the incomplete member syntax
+            var markup = @"
+class C
+{
+    partial int $$
+}
+";
+            await VerifySymbolKinds(markup, SymbolKind.Field, SymbolKind.Method, SymbolKind.Property);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async void LocalInsideMethod1()
+        {
+            var markup = @"
+namespace ConsoleApp1
+{
+    class ReallyLongClassName { }
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            ReallyLongClassName $$
+        }
+    }
+";
+            await VerifySymbolKinds(markup, SymbolKind.Local);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async void LocalInsideMethod2()
+        {
+            var markup = @"
+namespace ConsoleApp1
+{
+    class ReallyLongClassName<T> { }
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            ReallyLongClassName<int> $$
+        }
+    }
+";
+            await VerifySymbolKinds(markup, SymbolKind.Local);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async void LocalInsideMethodAfterPredefinedTypeKeyword()
+        {
+            var markup = @"
+namespace ConsoleApp1
+{
+    class ReallyLongClassName { }
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            string $$
+        }
+    }
+";
+            await VerifySymbolKinds(markup, SymbolKind.Local);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async void LocalInsideMethodAfterArray()
+        {
+            var markup = @"
+namespace ConsoleApp1
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            string[] $$
+        }
+    }
+";
+            await VerifySymbolKinds(markup, SymbolKind.Local);
+        }
+
+        private async Task VerifyNoType(string markup)
+        {
+            var result = await GetResultsAsync(markup);
+            Assert.Null(result.Type);
+        }
+
+        private async Task VerifyTypeName(string markup, string typeName)
+        {
+            var result = await GetResultsAsync(markup);
+            Assert.Equal(typeName, result.Type.Name);
```

Comment:
consider getting the full name.  That helps distringuish getting an error type symbol and a real type symbol.